### PR TITLE
some docs updated

### PR DESCRIPTION
**Pull request by Commandserver**
added [discord's keyword matching strategies](https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies) to `dpp::automod_metadata::keywords`

maked the second param of utility::timestamp optional!

### CODE REVIEW

File: include/dpp/automod.h [hunk 1]
```diff
@@ -102,7 +102,39 @@ enum automod_trigger_type : uint8_t {
...
+     * | \*cat*     | lo<u><b>cat</b></u>ion, edu<u><b>Cat</b></u>ion |
+     * | \*the mat* | brea<u><b>the mat</b></u>ter          |
+     *
+     * Whole Word - keyword is a full word or phrase and must be surrounded by whitespace at the beginning and end
+     *
+     * | keyword | matches     |
+     * |---------|-------------|
+     * | cat     | <u><b>Cat</b></u>     |
+     * | the mat | <u><b>the mat</b></u> |
+     *
```
Comment (Commandserver) [inline include/dpp/automod.h:137]: turns out like this:
image

File: src/dpp/message.cpp [hunk 2]
```diff
@@ -725,10 +725,10 @@ attachment::attachment(struct message* o, json *j) : attachment(o) {
 
 void attachment::download(http_completion_event callback) const {
 	/* Download attachment if there is one attached to this object */
-	if (!owner->owner) {
+	if (owner == nullptr || owner->owner == nullptr) {
```
Comment (Commandserver) [inline src/dpp/message.cpp:728]: the attachment's owner-attribute can also be nullptr

File: src/dpp/message.cpp [hunk 3]
```diff
@@ -725,10 +725,10 @@ attachment::attachment(struct message* o, json *j) : attachment(o) {
...
 		throw dpp::logic_exception("attachment has no owning message/cluster");
 	}
-	if (callback && this->id && !this->url.empty()) {
+	if (callback && !this->url.empty()) {
```
Comment (Commandserver) [inline src/dpp/message.cpp:731]: idk for what this was o:

File: include/dpp/utility.h [hunk 4]
```diff
@@ -89,18 +89,18 @@ namespace dpp {
...
 		/**
-		 * @brief Return a mentionable timestamp (used in a discord embed). These timestamps will display the given timestamp in the user's timezone and locale.
+		 * @brief Return a mentionable timestamp (used in a message). These timestamps will display the given timestamp in the user's timezone and locale.
 		 * 
 		 * @param ts Time stamp to convert
 		 * @param tf Format of timestamp using dpp::utility::time_format
 		 * @return std::string The formatted timestamp
 		 */
-		std::string DPP_EXPORT timestamp(time_t ts, time_format tf);
+		std::string DPP_EXPORT timestamp(time_t ts, time_format tf = tf_short_datetime);
```
Comment (Commandserver) [inline include/dpp/utility.h:98]: defaults now to the same as discord do: https://discord.com/developers/docs/reference#message-formatting-timestamp-styles

File: src/dpp/message.cpp (hunk 3 above)
Comment (braindigitalis) [reply src/dpp/message.cpp:731]: it checks there's an id for the attachment, you can't download without one. the check needs to remain...

**Review by braindigitalis — CHANGES_REQUESTED** [review]

File: src/dpp/message.cpp (hunk 3 above)
Comment (Commandserver) [reply src/dpp/message.cpp]: okay. its now back in

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**